apps/charterafrica/src/pages/api/v1/resources/ecosystem/[action].page.js

```diff
@@ -1,25 +1,49 @@
-import { schema } from "@/charterafrica/lib/ecosystem/airtable.old";
+import airtable from "@/charterafrica/lib/ecosystem/airtable";
 import {
-  updateEcosystemList,
-  updateEcosystemContent,
+  updateList,
+  updateContent,
 } from "@/charterafrica/lib/ecosystem/ecosystem";
 
 const isApiKeyValid = (key) => {
   return key && key === process.env.RESOURCES_SECRET_TOKEN;
 };
 
+async function schema(req, res) {
+  const { baseId } = req.query;
+  if (!baseId) {
+    return res.status(400).json({ message: "MISSING BASE ID" });
+  }
+  return airtable.schema(baseId);
+}
+
+async function bases() {
+  return airtable.bases();
+}
+
+async function entities(req, res) {
+  if (req.method === "POST") {
+    return updateList();
+  }
+  if (req.method === "PUT") {
+    return updateContent();
+  }
+  return res.status(405).json({ message: "METHOD_NOT_ALLOWED" });
+}
+
 const actionMap = {
-  "update-list": updateEcosystemList,
-  "update-content": updateEcosystemContent,
   schema,
+  bases,
+  entities,
 };
 
+const apiKeyExcludedActions = ["schema", "bases"];
+
 export default async function handler(req, res) {
   const {
     query: { action },
   } = req;
   const key = req.headers["x-api-key"];
-  if (!isApiKeyValid(key) && action !== "schema") {
+  if (!isApiKeyValid(key) && !apiKeyExcludedActions.includes(action)) {
```

Comment:
Looks like we only need this key in `entities`. Can't we move the check for key there?